src/turn_integrator.py

```diff
@@ -25,22 +26,42 @@ def turn_credentials(line_name):
 
     return eval_credentials(config_json)
 
-### Obtain a new auth token from Turn.io
-### TODO Not tested yet
-def obtain_auth_token(username, password):
-    u_pass = HTTPBasicAuth(username, password)
-    return requests.post('https://whatsapp.turn.io/v1/users/login', auth=u_pass)
+### CONTACTS
+## Obtain a contact profile. See documentation here:
+## https://whatsapp.turn.io/docs/api/contacts#retrieve-a-contact-profile
+def obtain_contact_profile(msisdn: str, line_name: str) -> requests.Response:
+    auth_headers = {
+        'Authorization': f'Bearer {turn_credentials(line_name)}',
+        'Accept': 'application/vnd.v1+json'
+    }
+
+    response = requests.get(f'https://whatsapp.turn.io/v1/contacts/{msisdn}/profile', headers=auth_headers)
+    print(response.text)
+    return response
+
+## Update a contact profile. Supply only the fields that need updating. See documentation here:
+## https://whatsapp.turn.io/docs/api/contacts#update-a-contact-profile
+def update_contact_profile(msisdn: str, line_name: str, profile_data: json) -> requests.Response:
+    auth_headers = {
+        'Authorization': f'Bearer {turn_credentials(line_name)}',
+        'Accept': 'application/vnd.v1+json'
+    }
+
+    response = requests.patch(f'https://whatsapp.turn.io/v1/contacts/{msisdn}/profile', headers=auth_headers, json=profile_data)
+    print(response.text)
+    return response
 
-### Send the different kinds of messages. See documentation here:
-### https://whatsapp.turn.io/docs/api/messages
-def send_message(message_data, line_name):
+### MESSAGES
+## Send the different kinds of messages. See documentation here:
+## https://whatsapp.turn.io/docs/api/messages
+def send_message(line_name: str, message_data: str) -> requests.Response:
```

Comment:
We should probably note that the `message_data` here is expected to be a json string. Probably in the docstrings.

Is the intention to later support the different types of messages more directly?

Reply:
Aaaahhh, type checking is not enforced in Python, interesting... Not sure I like that!

Reply:
Yup, no compiler does that to a language.